cloud/ec2.go

```diff
@@ -687,6 +689,36 @@ func (m *ec2Manager) SpawnHost(ctx context.Context, h *host.Host) (*host.Host, e
 				"host_provider": h.Distro.Provider,
 				"distro":        h.Distro.Id,
 			}))
+			if h.ShouldFallbackToOnDemand() && (strings.Contains(err.Error(), EC2InsufficientCapacity) || strings.Contains(err.Error(), EC2UnfulfillableCapacity)) {
+				// Spot creation failed due to an InsufficientCapacity error.
+				// Try to spawn the host as OnDemand instead of Spot
+				event.LogHostFallback(h.Id)
+				grip.Info(message.Fields{
+					"message":  "could not spawn spot instance, falling back to on-demand",
+					"host_id":  h.Id,
+					"host_tag": h.Tag,
+					"distro":   h.Distro.Id,
+					"provider": h.Provider,
+				})
+				h.Provider = evergreen.ProviderNameEc2OnDemand
+				h.Distro.Provider = evergreen.ProviderNameEc2OnDemand
+				if err = m.spawnOnDemandHost(ctx, h, ec2Settings, blockDevices); err != nil {
+					grip.Error(message.WrapError(err, message.Fields{
+						"message":       "spawning on-demand host",
+						"host_id":       h.Id,
+						"host_provider": h.Distro.Provider,
+						"distro":        h.Distro.Id,
+					}))
+					return nil, errors.Wrapf(err, "fallback to on-demand failed for host '%s'", h.Id)
```

Comment:
Starting a new thread since the last one is now out-dated. 

On second thought, it is more appropriate for this message to just say `spawning on-demand host '%s'` since we have already logged that we are falling back to on-demand for this specific host. The error message convention is to simply use the gerund of the failed action; previously we would say `problem spawning host...` or `error spawning host..` but found that since these are already error messages it was redundant to say `problem` or `error`. This redundancy is exacerbated since we often wrap errors which leads to a chain of error messages starting from the bottom of the call stack. You will notice some inconsistency in error messages and overall style, especially in older code. Kim recently made a huge effort to make our error messages uniform across all of our code bases but there still may be some inconsistency in Evergreens since it is so big.

Reply:
That's helpful context about the wrapping.

Reply:
Fixed